src/transaction_sender/transaction_pool.zig

```diff
@@ -0,0 +1,112 @@
+const std = @import("std");
+const network = @import("zig-network");
+const sig = @import("../sig.zig");
+
+const Allocator = std.mem.Allocator;
+const AtomicBool = std.atomic.Value(bool);
+const AtomicSlot = std.atomic.Value(Slot);
+
+const Slot = sig.core.Slot;
+const RwMux = sig.sync.RwMux;
+const Signature = sig.core.Signature;
+const TransactionInfo = sig.transaction_sender.TransactionInfo;
+
+/// Pool to keep track of pending transactions, transactions are added to the
+/// pool after they are received off the incomming channel, and removed when
+/// they are confirmed, failed, timed out, or reach the max retries.
+/// pending_transactions requires a lock as it is accessed by both the receive
+/// thread and the process transactions thread.
+/// retry_signatures and drop_signatures do not need a lock as they are
+/// only accessed by the process transactions thread.
+pub const TransactionPool = struct {
+    pending_transactions: RwMux(PendingTransactions),
```

Comment:
probs be worth suffixing with `_rw` (ie, `pending_transactions_rw`) so you can do `const pending_transactions, var lock = self.pending_transactions_rw.readWithLock();`

thats the main reason we use `_rw` throughout gossip